persistence/persistence-mybatis/src/main/java/org/cbioportal/persistence/mybatis/GeneMyBatisRepository.java

```diff
@@ -31,29 +31,84 @@
 */
 package org.cbioportal.persistence.mybatis;
 
-import java.util.ArrayList;
-import java.util.List;
 import org.cbioportal.model.Gene;
+import org.cbioportal.model.meta.BaseMeta;
 import org.cbioportal.persistence.GeneRepository;
+import org.cbioportal.persistence.PersistenceConstants;
+import org.cbioportal.persistence.mybatis.util.OffsetCalculator;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Repository;
 
-/**
- *
- * @author jiaojiao
- */
+import java.util.List;
+
 @Repository
-public class GeneMyBatisRepository implements GeneRepository{
+public class GeneMyBatisRepository implements GeneRepository {
+
     @Autowired
     GeneMapper geneMapper;
-    
+    @Autowired
+    private OffsetCalculator offsetCalculator;
+
+    @Override
+    public List<Gene> getAllGenes(String projection, Integer pageSize, Integer pageNumber, String sortBy,
+                                  String direction) {
+
+        return geneMapper.getGenes(projection, pageSize, offsetCalculator.calculate(pageSize, pageNumber), sortBy,
+                direction);
+    }
+
+    @Override
+    public BaseMeta getMetaGenes() {
+
+        return geneMapper.getMetaGenes();
+    }
+
+    @Override
+    public Gene getGeneByEntrezGeneId(Integer entrezGeneId) {
+
+        return geneMapper.getGeneByEntrezGeneId(entrezGeneId, PersistenceConstants.DETAILED_PROJECTION);
+    }
+
+    @Override
+    public Gene getGeneByHugoGeneSymbol(String hugoGeneSymbol) {
+
+        return geneMapper.getGeneByHugoGeneSymbol(hugoGeneSymbol, PersistenceConstants.DETAILED_PROJECTION);
+    }
+
+    @Override
+    public List<String> getAliasesOfGeneByEntrezGeneId(Integer entrezGeneId) {
```

Comment:
I'm not positive about how this is used .. but I suspect that the main use case for aliases is to do the reverse of what is supported here. I think the reason we need the aliases is so that when a user enters a gene name into the query box we can go from alias string to the gene object. That functionality is not provided here. Perhaps the use case for this "alias to gene" request is buried in servlets that we have not yet tried to replace. We should seek comments from others ( @jjgao @n1zea144 ?) . 
It is ok to go forward with this as is and add the needed functionality later .. but good to contemplate the need now, while we are adjusting the model.